cmd/tugger/main.go

```diff
@@ -211,23 +216,43 @@ func mutateAdmissionReviewHandler(w http.ResponseWriter, r *http.Request) {
 	w.Write(data)
 }
 
-func handleContainer(container *v1.Container, dockerRegistryUrl string) bool {
-	log.Println("Container Image is", container.Image)
+// imageExists verifies an image exists in the remote registry
+func imageExists(image string) bool {
+	ref, err := name.ParseReference(image)
+	if err != nil {
+		log.Print(err)
+		return false
+	}
 
-	if !containsRegisty(whitelistedRegistries, container.Image) {
-		message := fmt.Sprintf("Image is not being pulled from Private Registry: %s", container.Image)
-		log.Printf(message)
+	if _, err := remote.Image(ref, remote.WithAuthFromKeychain(authn.DefaultKeychain)); err != nil {
```

Comment:
The unit tests will fail if dockerhub is unreachable - can you wrap remote.Image() behind an interface to allow a mock to be injected for unit-testing purposes?

Reply:
Sure, mock added.